src/attribute-based/ABE_Utils.c

```diff
@@ -0,0 +1,68 @@
+#include "attribute-based/ABE_Utils.h"
+#include <stdlib.h>
+
+int Lagrange_coefficient(int xi, int* S, int sLength, int x)
+{
+    double result = 1;
+    for(int i = 0; i < sLength; i++)
+    {
+        if(&S[i] != NULL)
+        {
+            if(S[i] != xi)
+            {
+                double xD = (double) x;
+                double j = (double) S[i];
+                double xiD = (double) xi;
+                result = result*((xD-j)/(xiD-j));
+            }
+        }
+    }
+    return (int) result;
```

Comment:
It is needed, that was a bug fixed lately, for example Lagrange(1, [1, 2, 3], 0) returning 2 instead of returning 3 because of the subparts of the calculation that are not always integers (for example (xD-j)/(xid-j) is not always an integer, but result finally is always an integer, also the inputs are always integers).

I also made a test for a proof that the return value is always int:
`double m_Lagrange_coefficient(int xi, int* S, int sLength, int x)
{
    double result = 1;
    for(int i = 0; i < sLength; i++)
    {
        if(&S[i] != NULL)
        {
            if(S[i] != xi)
            {
                double xD = (double) x;
                double j = (double) S[i];
                double xiD = (double) xi;
                result = result*((xD-j)/(xiD-j));
            }
        }
    }
    return result;
}

void makeTest()
{
    for(int i = 2; i < 100; i++)
    {
        int indexes[i];
        for(int x = 0; x < i; x++)
        {
            indexes[x] = x+1;
        }
        for(int o = 1; o <= i; o++)
        {
            double res = m_Lagrange_coefficient(o, indexes, i, 0);
            if(ceilf(res) != floorf(res))
            {
                printf("%f\n", res);
            }
        }
    }
}`
code does not print any value. (Regarding the ABE part of the library, it does only use this method with i <= 32.)